### PR TITLE
[Feat] More control over point label

### PR DESCRIPTION
**Pull request by heshan0131**
- Allow adding multiple labels
- Allow edit label alignment
- Offset label by point radius

Screen Shot 2019-05-17 at 3 17 37 PM

### CODE REVIEW

**Comment by heshan0131** [comment]
#497

File: examples/demo-app/src/app.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ class App extends Component {
     //   window.setTimeout(this._showBanner, 3000);
     // }
     // load sample data
-    // this._loadSampleData();
+    this._loadSampleData();
```
Comment (macrigiuseppe) [inline examples/demo-app/src/app.js]: comment this one

File: src/actions/vis-state-actions.js [hunk 2]
```diff
@@ -37,6 +37,26 @@ export function layerConfigChange(oldLayer, newConfig) {
   };
 }
 
+/**
+ * Update layer text label
+ * @memberof visStateActions
+ * @param {Object} oldLayer - layer to be updated
```
Comment (macrigiuseppe) [inline src/actions/vis-state-actions.js]: why using the name __oldLayer__?
wouldn't be better __layer__?

Comment (macrigiuseppe) [reply src/actions/vis-state-actions.js]: I see, you are calling layer config updater

File: src/layers/point-layer/point-layer.js [hunk 3]
```diff
@@ -297,6 +319,27 @@ export default class PointLayer extends Layer {
...
+    const yMult = config.alignment === 'center' ? 0 :
+      config.alignment === 'bottom' ?  1 : -1;
+
+    const sizeOffset = config.alignment === 'center' ? 0 :
+    config.alignment === 'bottom' ?  config.size : config.size;
+
+    const pixelRadius = radiusScale * distanceScale.pixelsPerMeter[0];
+
+    return typeof getRadius === 'function' ? d => [
+      xMult * (getRadius(d) * pixelRadius + 20),
```
Comment (macrigiuseppe) [inline src/layers/point-layer/point-layer.js]: Define a constant for __20__

File: src/layers/point-layer/point-layer.js [hunk 4]
```diff
@@ -297,6 +319,27 @@ export default class PointLayer extends Layer {
...
 
+  getTextOffset(config, radiusScale, getRadius, mapState) {
+    const distanceScale = getDistanceScales(mapState);
+    const xMult = config.anchor === 'middle' ? 0 :
+      config.anchor === 'start' ? 1 : -1;
+    const yMult = config.alignment === 'center' ? 0 :
+      config.alignment === 'bottom' ?  1 : -1;
+
+    const sizeOffset = config.alignment === 'center' ? 0 :
+    config.alignment === 'bottom' ?  config.size : config.size;
```
Comment (macrigiuseppe) [inline src/layers/point-layer/point-layer.js]: add space at the beginning of the statement to be consistent with the rest of the method

**Comment by macrigiuseppe** [comment]
Running the demo, I noticed the distance between label and point changes based on the zoom level but is odd when map is particular zoomed in or out. See pictures below:

__Zoomed In__
Screenshot 2019-05-17 21 35 13

__Zoomed Out__
Screenshot 2019-05-17 21 35 25

File: examples/demo-app/src/app.js (hunk 1 above)
Comment (heshan0131) [reply examples/demo-app/src/app.js]: done

File: src/layers/point-layer/point-layer.js (hunk 3 above)
Comment (heshan0131) [reply src/layers/point-layer/point-layer.js]: done

File: src/layers/point-layer/point-layer.js (hunk 4 above)
Comment (heshan0131) [reply src/layers/point-layer/point-layer.js]: done

**Pull request merged**